### ISSUE TITLE
Turn off import/no-absolute-path

### ISSUE THREAD

**Issue by HoraceShmorace** [standard/standard#1535]
**What version of this package are you using?**
14.3.1

**What problem do you want to solve?**
This rule can't be used with projects that involve AWS Lambda Functions that reference custom code in Lambda Layers. Layer code MUST be imported relative to the root (`'/opt/[layer path]'`), which is an absolute path.

I've finally gotten our teams to use Standard for almost every project, zero customizations. Now I'm getting pushback because of this issue. We could disable that rule inline, but if you have to turn a rule off 20 times, then is it a worthwhile rule?

I'm sure someone will suggest that we just figure out the relative path (even though AWS's official docs say to use the absolute path), but then we'll be making assumptions about the current and (more importantly) future state of a filesystem we have no control over, and could likely change, breaking our services.

**What do you think is the correct solution to this problem?**
Turn this rule off. If it can't be a best practice everywhere, then it's not worthwhile.

**Are you willing to submit a pull request to implement this change?**
Sure

**Comment by HoraceShmorace** [comment]
I'll move this to the eslint-config-standard repo.

**Comment by mightyiam** [comment]
Thank you, @HoraceShmorace. Actually, this is the repository where rules are discussed.

This seems like it would be a popular enough use case to consider disabling this rule. I personally would not like to go through that documentation. But I would like the issue confirmed.

**Comment by 75lb** [comment]
I too would prefer this rule is removed from standard. What is the problem with absolute paths anyway? They are valid.

In my web projects, this rule prevents me importing npm-installed ES modules like this: 
```
import something from '/node_modules/something/index.mjs'
```

Instead, the rule forces me to use messy and brittle relative paths (which need fixing if the source file changes directory):

```
import something from '../../../../../node_modules/something/index.mjs'
```

It is common for absolute paths to be used in web projects (for example `<link rel="stylesheet" href="/global.css">)`, I don't see the logic for blocking their use in `import` statements.

**Comment by feross** [comment]
I agree that this is a bit prescriptive, but this doesn't seem like a common issue so it seems like a perfect use case for using a disable comment as explained here: https://standardjs.com/#how-do-i-disable-a-rule

How many others have run into this issue?

**Comment by feross** [comment]
I just realized that this rule is applying for browser imports too. Given that, I feel more strongly that this should be disabled now. We wouldn't want to prevent referring to a module file by absolute paths. That's totally fine in a browser context.

**Comment by audinue** [comment]
> I just realized that this rule is applying for browser imports too. Given that, I feel more strongly that this should be disabled now. We wouldn't want to prevent referring to a module file by absolute paths. That's totally fine in a browser context.

What about an option to choose a context instead of disabling this rule completely?

This rule maybe helpful in Node.js context but certainly not in browser context.

> Node.js allows the import of modules using an absolute path such as /home/xyz/file.js. That is a bad practice as it ties the code using it to your computer, and therefore makes it unusable in packages distributed on npm for instance.

From: https://github.com/benmosher/eslint-plugin-import/blob/master/docs/rules/no-absolute-path.md

**Comment by feross** [comment]
> What about an option to choose a context instead of disabling this rule completely?

Reasonable suggestion, but we don't want to add options like this to `standard`. Having to specify if your file is meant for Node vs. Browser is not in the sprit of `standard`. We want `standard` to get out of the user's way.

**Comment by AbducteeZero** [comment]
This rule should simply be deleted, and whoever put it in place needs to be removed from such decision making.  It is clearly technically incorrect.

**Comment by LinusU** [comment]
@AbducteeZero you comment doesn't add anything of value at all, and instead just insults the people working on this project. Clearly there was grounds for adding the rule when it was added, and now more information have come to light and it it seems like removing it is the best path forward....

**Comment by AbducteeZero** [comment]
> This rule should simply be deleted, and whoever put it in place needs to be removed from such decision making. It is clearly technically incorrect.



> @AbducteeZero you comment doesn't add anything of value at all, and instead just insults the people working on this project. Clearly there was grounds for adding the rule when it was added, and now more information have come to light and it it seems like removing it is the best path forward....

Oops, sorry for being an engineer that states facts and wants to take action based on those facts in an effort to move forward.  Yep, I am attacking wrong action, so that right action may continue instead.  Get your feelings out of programmer discussion.

**Comment by obonyojimmy** [comment]
i think this is a valid issue affecting lots of devels and needs to be resolved .

**Comment by derek90** [comment]
What about adding some config, where you can specify to skip this rule for a specific path tamplate, like `'/opt/**'`. That way, you can keep that rule for other cases where you are able to change the absolute path for a relative one

**Comment by 75lb** [comment]
Is there any known workaround for this yet? I still can't run `standard --fix` on my web/browser projects without breaking all the absolute path imports..

**Comment by HoraceShmorace** [comment]
Wow, this is still unresolved? I ended up just overriding the rule, and then at some point my engineers argued that if we're overriding the "standard" then it's not a standard, and we moved back to straight ESLint. Standard's implementation of this rule was always too opinionated to be a standard.

**Comment by 75lb** [comment]
I personally consider it a bug as it breaks code. If you clone this [reproduction case](https://gist.github.com/75lb/2d86f6327e4c7308ddf05d5e311c6800) then run `standard --fix index.js` within the repo directory, you will see the following `index.js` line...

```js
/* in browser context, `/` means "the root directory of the web site" */
import something from '/node_modules/something/index.js'
```

...change to the following, which points to an entirely different location (the `/` root folder of the device):

```js
import something from '../../../../../node_modules/something/index.js'
```

As it stands, `standard` is ***incompatible with web/browser projects*** as it seems unable to detect browser context, (where `standard --fix` breaks `import` paths).

**Comment by voxpelli** [comment]
> standard is incompatible with web/browser projects

Rather: It expects paths to be to local files and to have a bundler / compile step that translates that to whatever paths it will be on the server (which will also deal with cache busting etc)

Blanket statements helps no one.

**Comment by 75lb** [comment]
I thought this tool is called "standard" - if a DSL requires a compile step to convert it to Javascript then by definition it is not "standard".

**Comment by voxpelli** [comment]
@75lb https://github.com/standard/standard?tab=readme-ov-file#but-this-isnt-a-real-web-standard
